pkg/menu/menu.go

```diff
@@ -19,23 +32,42 @@ func newParagraph(initText string, border bool, location int, wid int, ht int) *
 	return p
 }
 
-func processInput(input, warning *widgets.Paragraph, isValid func(string) (string, bool), uiEvents <-chan ui.Event) (string, error) {
-	// keep tracking all input from user
+// readKey reads a key from input stream.
+func readKey(uiEvents <-chan ui.Event) string {
 	for {
 		e := <-uiEvents
-		if e.Type != ui.KeyboardEvent {
-			continue
+		if e.Type == ui.KeyboardEvent {
+			return e.ID
 		}
-		switch e.ID {
+	}
+}
+
+// processInput presents an input box to user and returns the user's input.
+// processInput will check validation of input using isValid function.
+func processInput(introwords string, location int, wid int, ht int, isValid func(string) (string, string, bool), uiEvents <-chan ui.Event) (string, string, error) {
```

Comment:
Maybe we can make a type `validCheck` or something for the isValid function type so you dont have to keep typing the full thing